### DISCUSSION TITLE
Association support

### DISCUSSION THREAD

**Discussion by lorefnon** [komapper/komapper#541] [Ideas]
In conventional ORMs (like ActiveRecord) we have support to define associations (eg. hasMany, hasOne, belongsTo, hasAndBelongsToMany) between entities and then the ORM is able to support lazy/eager fetching of associated entities while fetching entities. 

So I could do something like fetch all departments, users in those departments and their profiles in effectively a single line. 

Are there any plans to add similar support in komapper ? 

I understand I can manually construct joins and map the rows to multiple entities but it is quite convenient to have built in association support.

**Comment by nakamura-to** [comment]
We provide `include` and `includeAll` functions to fetch associated entities:

https://www.komapper.org/docs/reference/query/querydsl/select/#include
https://www.komapper.org/docs/reference/query/querydsl/select/#includeall

We do not plan to adopt lazy loading because we want to avoid the N + 1 query problem.

**Reply by lorefnon** [reply]
Oh ok, I missed the include support in documentation (wasn't looking in the right place). 

Thanks a lot. I understand the concern around N+1.
